test/IDE/complete_ambiguous.swift

```diff
@@ -425,3 +427,14 @@ CreateThings {
     }
     Thing. // ErrorExpr
 }
+
+
+func takesClosureOfPoint(_: (Point)->()) {}
+func overloadedWithDefaulted(_: ()->()) {}
+func overloadedWithDefaulted(_: ()->(), _ defaulted: Int = 10) {}
```

Comment:
The question - is there a single best solution could be answered by calling `ConstraintSystem::findBestSolution` on the set of solution, that would give you an index of the best solution. So I guess we could filter out all of solutions with holes and run `findBestSolution` on the remaining ones, that would determine whether there is a solution which is viable to be applied to AST and would avoid having to re-typecheck.

Reply:
That works for the defaulted case (it sees the non-defaulted one as a specialization of the other) but it doesn't account for the change we made for code completion to allow missing arguments, so the below gets no best solution:
```
func takesClosureOfPoint(_: (Point)->()) {}
func overloadedWithMissing(_: ()->()) {}
func overloadedWithMissing(_: ()->(), _ required: String) {}

takesClosureOfPoint { p in
  overloadedWithDefaultedOrMissing {
    if p.#^REGULAR_MULTICLOSURE_APPLIED^# {}
  }
}
```

Reply:
But in this case since it's a trailing closure second overload of `overloadedWithMissing` shouldn't match right?

Reply:
I guess we shouldn't allow missing arguments in this case since that would be syntactically incorrect.

Reply:
Yeah, it really only should if the missing arg is also a closure (since we allow multiple trailing closures now). The test case doesn't need to be trailing to have this problem though.

Reply:
Alright, I didn't realize that we'd have to type-check anyway so I think this is okay.